### ISSUE TITLE
Update to use Spring Boot 3 (3.0.5), Spring Framework 6 (6.0.7) and springdoc-openapi 2

### ISSUE THREAD

**Issue by yoomlam** [department-of-veterans-affairs/abd-vro#1428]
SEE MOST RECENT COMMENT FOR LATEST CONTEXT

Lighthouse DI [announced](https://dsva.slack.com/archives/C03UA9MV1EH/p1680803432357659) Starter-boot release 1.0.0 available now:
> This new release comes with Spring Boot 3 (3.0.5), Spring Framework 6 (6.0.7) and springdoc-openapi 2.  Packages are available [here](https://github.com/orgs/department-of-veterans-affairs/packages?repo_name=lighthouse-di-starter-boot). You can upgrade your project to use Spring Boot 3 if you wish to do so. If you have any questions please put comments in this thread.

This will keep VRO updated and could help resolve some SecRel alerts.

We'd need to do thorough testing since these are major version updates.
Also need to check Spring [version compatibility](https://github.com/department-of-veterans-affairs/abd-vro/blob/5f4d50d06999ba62519e4802afe762ff2d2defd9/gradle.properties#L17-L19) with Camel version, which should also be updated.

**Comment by yoomlam** [comment]
@msnwatson Any project that uses Camel may have to stay with Spring Boot 2, until Camel 4 (which uses Spring Boot 3) is released https://camel.apache.org/releases/.

**Comment by dianagriffin** [comment]
@msnwatson If this ticket won't be completed before the new sprint starts tomorrow, could you comment with a summary of where we're at on this, and what remains?

**Comment by msnwatson** [comment]
Code changes have been made and pushed to a feature branch to upgrade to Spring 6 and Spring Boot 3. However, Apache Camel 3 (the latest version) does not support Spring Boot 3.

Apache Camel 4 is a requirement for completing this ticket, and the Apache Camel team's roadmap shares that they expect the new version to be released in the month of May.

So, while significant progress has been made on this ticket, it cannot be completed until Apache Camel 4 is released.

**Comment by msnwatson** [comment]
Moving this ticket to the backlog, as there is some divergence from the planned Apache Camel 4 roadmap that has led to more uncertainty in the time until this ticket will be unblocked.

Estimated remaining time for the ticket's completion is 2 story points and feature branch containing the current work for this ticket is at https://github.com/department-of-veterans-affairs/abd-vro/tree/mwatson/spring6

**Comment by msnwatson** [comment]
Given the delay in the release of Camel 4, this ticket is being re-scoped so that it will be considered complete when:

1. All containers that do not make use of the Camel library are successfully upgraded and tested with the latest versions of Spring Boot 3, Spring 6 and springdoc-openapi2 (note that this may require some effort to disentangle the containers that do not use Camel from those that do)

A new ticket will be created to cover the work to upgrade containers which use Camel to these new spring versions in conjunction with upgrading to Camel 4.
